### PR TITLE
Issue #4181: Fix stub entities

### LINKED ISSUE #4181: Remove entity_create_stub_entity() in favor of entity_create()
In https://github.com/backdrop/backdrop-issues/issues/1475 I propose to add forward revision capability to core. The corresponding issue in D7 listed [Remove stub entities, replace entity_create_stub_entity()](https://www.drupal.org/project/drupal/issues/1551140) as a blocker, subsequently fixed. This issue is to decide if this needs to follow the D7 path.

`entity_create_stub_entity()` is called only in four places: 3 times in Field tests and also in `EntityFieldQuery::finishQuery()` and just creates a `stdClass()` Object to initialize a stub entity, which was fine when all D7 Entities were not classed, but now in Backdrop all Entities are classed Objects.

The original resolution to stub entities when D7 started classing entities was to just change  `entity_create_stub_entity()` 
```
function entity_create_stub_entity($entity_type, $ids) {
- $entity = new stdClass();
...
-  return $entity;
+ return isset($info['entity class']) ? entity_create($entity_type, $values) : (object) $values;
}
```

But ultimately the patch committed just before forward revisions was committed was not that "simple" fix but [this one](https://www.drupal.org/files/stub-entity-concept-removed-1551140-38.patch) which removes stub entities completely. 

This issue is to track which route would be best for Backdrop. 

Would appreciate some advice from @quicksketch on how he feels we should move forward as this as noted above is a blocker.

---
PR by @docwilmot https://github.com/backdrop/backdrop/pull/2966

### PR DESCRIPTION
**Pull request by docwilmot**
Fixes https://github.com/backdrop/backdrop-issues/issues/4181

### CODE REVIEW

**Review by herbdool — COMMENTED** [review]
This code looks good to me. I haven't tested but looks like it's mostly about tests passing.

File: core/modules/entity/entity.module [hunk 1]
```diff
@@ -349,18 +349,20 @@ function entity_extract_ids($entity_type, $entity) {
  *
  * @return
  *   An entity object, initialized with the IDs provided.
+ *
+ * @deprecated since 1.15.0.
  */
 function entity_create_stub_entity($entity_type, $ids) {
-  $entity = new stdClass();
```
Comment (quicksketch) [inline core/modules/entity/entity.module:354]: Could you add a deprecated watchdog call in here as the first line?
```
watchdog_deprecated_function('entity', __FUNCTION__);
```

Comment (docwilmot) [reply core/modules/entity/entity.module:354]: Done

**Comment by docwilmot** [comment]
Sandbox did funny things, so re-opening.

**Comment by quicksketch** [comment]
Thanks @docwilmot! Merged into 1.x for 1.15.0.

**Pull request merged**